### DISCUSSION TITLE
系统指令（system) 和用户指令 (prompt)

### DISCUSSION THREAD

**Discussion by chenxizhang** [chenxizhang/openai-powershell#186] [Use cases - 简体中文]
在 ChatGPT 的世界里面，提示词（prompt）是很重要的，它代表了我们作为用户与大语言模型进行交互的指令。而它又分为两类，一类是系统指令（本模块中通过 system 参数指定），一类是用户指令 （本模块中通过 prompt 参数指定）。

**系统指令**一般是角色定义，也就是我们系统大语言模型所扮演的角色，它的能力，以及你希望它在跟你对话或者帮助你生成内容时的一些规则。**用户指令**一般指的是具体你要它帮你做的事情，例如写一篇作文等等。

> [!TIP]
> 这两个参数是可以分开用，也可以混合用的。 system 参数会有默认值，所以无论你是否指定，其实它都有一个默认设置。而当你指定了prompt，就意味着你不是要聊天，而是要直接返回文本生成结果了。

他们都是字符串的形式，但在具体指定时你有多种选择。

## 直接输入

这是最简单的方式，你可以直接在调用命令时输入，显然它比较适合比较简单的场景，例如：

```powershell
# 以一个历史学家的角色来开启对话
chat -system "你是一个著名的历史学家，你精通中国上下五千年的历史，并且擅长用小学生听得懂的语言讲解历史，幽默风趣又不失端庄稳重。“

# 直接返回文本生成结果
gpt "中国的首都是哪一个城市?"

# 同时使用 system 和 prompt 两个参数
chat -system "你是一个著名的历史学家，你精通中国上下五千年的历史，并且擅长用小学生听得懂的语言讲解历史，幽默风趣又不失端庄稳重。“ `
 -prompt  "秦朝为什么会那么快灭亡呢"
```

## 文件输入

在调用命令时直接输入 system 或 prompt 当然是很直观的， 但是一个事实就是你的指令通常不会这么简单。这时候你就可以用文件来保存他们，然后将文件路径传递给命令。

```powershell

# 同时使用 system 和 prompt 两个参数, 本例中，为 system 参数传入的其实是一个文件名 expert.md，它会被自动读取。
gpt -system expert.md -prompt  "秦朝为什么会那么快灭亡呢"
```

## 在线网页

另外一种做法是直接读取一个网页地址，当作 system 或 prompt的输入，前提是你的这个文件是可以公开地访问到。

```powershell

# 同时使用 system 和 prompt 两个参数, 本例中，为 prompt 参数传入的其实是一个网址，它会被自动读取。
gpt -system “你帮我把文字从中文翻译成英文" `
 -prompt  "https://raw.githubusercontent.com/chenxizhang/openai-powershell/master/README.zh.md"
```

## 模板库

我更为得意的一个设计就是，你可以直接使用模板库的提示词，快速复用到你的代码中。当前我维护了一个提示词模板库（<https://github.com/code365opensource/promptlibrary>）, 里面已经有两个模板（fitness 是一个健身教练， gpt-translator-md 是可以把一个md文件进行翻译的模板）。

你可以像下面这样使用它们。

```powershell
# 同时使用 system 和 prompt 两个参数, 本例中，为 prompt 参数传入的其实是一个网址，它会被自动读取。
gpt -system “lib:gpt-translator-md" `
-prompt  "https://raw.githubusercontent.com/chenxizhang/openai-powershell/master/README.zh.md" `
-context @{from_language="zh";to_language="en"}
```

**Comment by xiaoyangliu1123** [comment]
陈老师，这里给的超链接有点问题，不小心包含了中文括号“）”。正确的链接或许是：
https://github.com/code365opensource/promptlibrary

**Reply by chenxizhang** [reply]
已更新
